npm/normalize-obstructive-user-agent/src/main.ts

```diff
@@ -0,0 +1,44 @@
+type Options = {
+  CypressUserAgentRe: string | RegExp
+  ElectronUserAgentRe: string | RegExp
+}
+
+const defaultOptions: Options = {
+  CypressUserAgentRe: /Cypress.*?\s/g,
+  ElectronUserAgentRe: /[Ee]lectron.*?\s/g,
+}
+
+export const setUserAgentOverride = async (replacementUserAgent: string) => {
+  try {
+    await Cypress.automation('remote:debugger:protocol', {
+      command: 'Network.setUserAgentOverride',
+      params: {
+        userAgent: replacementUserAgent,
+      },
+    })
+  } catch (e) {
+    throw new Error('Error occurred setting user agent')
+  }
+}
+
+export const replaceCypressInUserAgent = (userAgent: string, regularExp: string | RegExp = defaultOptions.CypressUserAgentRe): string => {
+  return userAgent.replace(regularExp, '')
+}
+
+export const replaceElectronInUserAgent = (userAgent: string, regularExp: string | RegExp = defaultOptions.ElectronUserAgentRe): string => {
+  return userAgent.replace(regularExp, '')
+}
+
+export const normalizeObstructiveUserAgent = async (options?: Partial<Options>) => {
```

Comment:
Shouldn't this hook into the `window:before:load` event so this is applied on page navigate and/or reloads or re-runs?

Reply:
I assume UA could impact both requests & scripts as they load on the page

Reply:
I don't think its an issue for navigation since the UA is set on top, but i'll explore something like `cy.reload` and see what happens. If that's an issue, we probs should handle it implicitly.